src/pages/index.js

```diff
@@ -27,71 +32,186 @@ const userinfoComponent = new Userinfo(
 );
 
 //Api
-const api = new Api({
-  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en/users/me ",
+
+const apiURL = {
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en/",
   headers: {
     authorization: "a1101938-3641-4790-a37b-6b7f03e0e338",
     "Content-Type": "application/json",
   },
-});
+};
 
-fetch("https://around.nomoreparties.co/v1/cohort-3-en/users/me", {
-  method: "PATCH",
-  headers: {
-    authorization: "c56e30dc-2883-4270-a59e-b2f7bae969c6",
-    // "Content-Type" "application/json"
+// Add Card Popup//
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  (cardData) => {
+    addCardPopup.showLoading();
+    api
+      .addNewCard(cardData)
+      .then((card) => {
+        renderCard(card);
+        addCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        addCardPopup.hideLoading();
+      });
   },
-  body: JSON.stringify({
-    name: "Marie Skłodowska Curie",
-    about: "Physicist and Chemist",
-  }),
-});
-// Card
-function handleCardImageClick(cardData) {
-  imagePreviewModal.open(cardData);
-}
+  "Saving..."
+);
 
-function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardImageClick);
-  const cardElement = card.getView();
-  section.addItem(cardElement);
+// Delete Card //
+const deleteCardPopup = new PopupWithConfirmation(
+  "#delete-card-modal",
+  "Deleting..."
+);
+deleteCardPopup.setEventListeners();
+
+function handleDeleteClick(card, cardID) {
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.showLoading();
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.removeCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.hideLoading();
+      });
+  });
+
+  deleteCardPopup.open();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+// Editing Profile //
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (inputValues) => {
+    editAvatarPopup.showLoading();
+
+    api
+      .updateProfilePicture(inputValues)
+      .then(() => {
+        profileAvatar.src = inputValues.link;
```

Comment:
Please create a method `setAvatar`, which will set the src, in UserInfo class and use it here

Reply:
Please fix this issue

Reply:
Use `  setUserImage` method of UserInfo class here